packages/ui/src/accounts/hooks/useGroupLocks/helpers.ts

```diff
@@ -46,12 +49,12 @@ export const useIsVoteLockRecoverable = (hasVoteLock: boolean, stakingAccount: s
     variables: { where: { castBy_eq: stakingAccount, stakeLocked_eq: true } },
     skip: !hasVoteLock,
   })
-  const { election: voteElection } = usePastElection(String(vote?.electionRound.cycleId ?? 0))
+  const { election: latestElection } = useLatestElection({ skip: !hasVoteLock })
+
+  if (!hasVoteLock || !vote || !latestElection) return
 
-  const isElectionFinished = !!voteElection
-  const { election: latestElection } = useLatestElection({ skip: !hasVoteLock || !isElectionFinished })
-  const isInLatestElection = !isElectionFinished || latestElection?.cycleId === voteElection?.cycleId
   // Always recoverable when the vote was not cast during the latest election
   // Otherwise recoverable when the election is over and the voted for candidate was not elected
-  return !isInLatestElection || (isElectionFinished && vote?.voteFor?.status !== CandidacyStatus.Elected)
+  const isInLatestElection = latestElection.cycleId === vote.electionRound.cycleId
```

Comment:
does it make a difference that only the first vote is checked? (`{ castVotes: [vote] }`)